### ISSUE TITLE
[bun-test] `$variable` does not work in `describe.each`

### ISSUE THREAD

**Issue by mProjectsCode** [oven-sh/bun#5752]
### What version of Bun is running?

1.0.2+37edd5a6e389265738e89265bcbdf2999cb81a49

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

Write a test like this
```ts
describe.each([
	{
		input: 'abc',
                expected: '',
	},
])(`test $input`, (data) => {
	// do something
});
```

### What is the expected behavior?

`test abc` as the test name, as described here https://jestjs.io/docs/api#describeeachtablename-fn-timeout.

### What do you see instead?

`test $input` as the test name.

**Comment by fightingdreamer** [comment]
Vitest also support this [format](https://vitest.dev/api/#test-each) in `test.each` and `describe.each`.

Moving to `bun test` without this feature is no-go, cause thousands of tests need to be manually migrated.

**Comment by ghost** [comment]
I've done a little bit of looking into this. I'll post some info below:

- The example in OP uses backticks for the test title. This could be simplified using just a regular string, so as to not be confused with template literal placeholders (`${expression}`).
- Jest uses an either/or approach when choosing to format `printf` or `$variable`. It prefers `printf` over `$variable` when doing an initial pass-through of the test title. So if a `printf` placeholder is found in the title that is not and escaped placeholder (`%%`), `printf` formatting will be done.
    - Bun could possibly choose to support both at the same time. This could be an optimization over how Jest is doing it, and could remove the need to do an initial pass-through. Just a thought that came up.
- `$variable` formatting supports basic object key paths (e.g., `$a.b.c`) and the index of the test case (`$#`, which is the same as `%#` in `printf` format).

The above may be helpful when defining the acceptance criteria for this issue.

**Comment by mProjectsCode** [comment]
My example above worked in Jest when I opened the issue. Bun aiming for Jest compatibility should be enough reason to care about this issue, as formulated by me.

**Comment by ghost** [comment]
> My example above worked in Jest when I opened the issue. Bun aiming for Jest compatibility should be enough reason to care about this issue, as formulated by me.

True, the example probably does work. My apologies on stating the example was incorrect. I'll correct my message above 👍  Backticks aren't necessary for this to work, so the example is more complex than it could be. Backticks allow placeholders (`${expression}`) to be used, but `$variable` is not a valid placeholder, nor is it being used by Jest as a template literal placeholder. So, perhaps the example could be simplified by using just normal strings.

**Comment by tmr232** [comment]
For those interested, I found a workaround for this:

1. Generate a map of `test-name -> test parameters`
2. Pass the map's keys (`[...myMap.keys()]`) to `test.each`/`describe.each`
3. Use `"%s"` for the label
4. In the test case, use the name to get the args from the map

It ain't pretty, but it seems to work.

**Comment by pere-seapoint** [comment]
Plus one on getting this prioritised. Thanks!

**Comment by orimay** [comment]
And one more! The only thing holding me back from switching over Bun

**Comment by dylan1951** [comment]
This makes me sad :(
